### PR TITLE
GitHub client repo branch

### PR DESCRIPTION
**Pull request by ncphillips**
Add repo and branch names to client.

Satisfies DIP by allowing components and hooks to indirectly call the `getForkRepoName` and `getHeadBranch` functions which grab them from cookies. 

The GithubClient know knows the repo name and branch name for both the fork and the base

### CODE REVIEW

File: packages/react-tinacms-github/src/form/useGithubFileForm.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ export const useGithubFileForm = <T = any>(
     fields: formOptions.fields || [],
     // save & commit the file when the "save" button is pressed
     onSubmit(formData) {
+      const github: GithubClient = cms.api.github
       return cms.api.github
```
Comment (dwalkr) [inline packages/react-tinacms-github/src/form/useGithubFileForm.ts:43]:
```suggestion
      return github
```

File: packages/react-tinacms-github/src/github-client/index.ts [hunk 2]
```diff
@@ -73,7 +72,18 @@ export class GithubClient {
     })
   }
 
-  async fetchExistingPR(forkRepoFullName: string, headBranch: string) {
+  get repoFullName(): string {
+    return getForkName() || this.baseRepoFullName
```
Comment (jamespohalloran) [inline packages/react-tinacms-github/src/github-client/index.ts:76]: I could see this maybe affecting behaviour in the short-term (If your fork cookie gets deleted, we might start committing to the base repo)

File: packages/react-tinacms-github/src/github-client/index.ts [hunk 3]
```diff
@@ -73,7 +72,18 @@ export class GithubClient {
...
+  }
+
+  get branchName() {
+    return getHeadBranch()
+  }
+
+  async fetchExistingPR() {
+    const forkRepoFullName = this.repoFullName
```
Comment (jamespohalloran) [inline packages/react-tinacms-github/src/github-client/index.ts:84]: misleading `forkRepoFullName` variable if it falls back to  `getForkName() || this.baseRepoFullName`

**Comment by jamespohalloran** [comment]
If we're moving the cookie business inside The GIthubClient, I think [these helpers](https://github.com/tinacms/tinacms/blob/github-client-repo-branch/packages/react-tinacms-github/src/github-editing-context/repository.ts) should be private functions on the GIthubClient.

We're currently calling `setForkName` inside the `GithbubAuthModal`. This would cause the GithubClient to have an out of sync forkName

File: packages/react-tinacms-github/src/github-client/index.ts (hunk 2 above)
Comment (ncphillips) [reply packages/react-tinacms-github/src/github-client/index.ts:76]: Very true, I'll make it throw instead

**Comment by ncphillips** [comment]
> If we're moving the cookie business inside The GIthubClient, I think these helpers should be private functions on the GIthubClient.

Great! Thanks, I'm going to change that

> We're currently calling setForkName inside the GithbubAuthModal. This would cause the GithubClient to have an out of sync forkName

I don't think this would be a problem. `repoFullName` is a getter, it fetches the value from the cookies on every request.

File: packages/react-tinacms-github/src/github-client/index.ts [hunk 4]
```diff
@@ -71,9 +74,31 @@ export class GithubClient {
         base: this.baseBranch,
       },
     })
+
+    this.setCookie(GithubClient.FORK_COOKIE_KEY, pullRequest.full_name)
```
Comment (jamespohalloran) [inline packages/react-tinacms-github/src/github-client/index.ts:78]: Did you mean to put this in the `createFork` function?

Comment (ncphillips) [reply packages/react-tinacms-github/src/github-client/index.ts:78]: Yes that's how it was being used before.

Comment (ncphillips) [reply packages/react-tinacms-github/src/github-client/index.ts:78]: But it was being done in [the modal](https://github.com/tinacms/tinacms/blob/master/packages/react-tinacms-github/src/github-editing-context/GithubAuthModal.tsx#L85-L86)

Comment (ncphillips) [reply packages/react-tinacms-github/src/github-client/index.ts]: Nope I'm tripping out

**Pull request merged**